compiler/src/pretyping.ml

```diff
@@ -2128,6 +2236,29 @@ let tt_global pd (env : 'asm Env.env) _loc (gd: S.pglobal) : 'asm Env.env =
   Env.Vars.push_global env (x,d)
 
 
+let tt_typealias arch_info env id ty = 
+  let ty = L.unloc ty in
+  match ty with 
+  | S.TBool -> Env.TypeAlias.push env id P.tbool
+  | S.TInt -> Env.TypeAlias.push env id P.tint
+  | S.TWord ws -> Env.TypeAlias.push env id (P.Bty (P.U ws))
+  | S.TArray (ws,e)-> 
+    begin
+      let arr = match ws with
+      | TypeWsize ws -> P.Arr (arch_info.pd, fst (tt_expr ~mode:`OnlyParam ws env e))
+      |TypeSizeAlias alias -> 
+        begin
+        let talias = Env.TypeAlias.get env alias in 
+        match L.unloc talias with 
+        | P.Bty (P.U ws) -> P.Arr (arch_info.pd, fst (tt_expr ~mode:`OnlyParam ws env e))
+        | _ -> rs_tyerror  ~loc:(L.loc id) (InvalidTypeAlias ((L.unloc id), L.unloc talias))
+        end 
+      in Env.TypeAlias.push env id arr
+    end
+  | S.TAlias alias_id -> 
+    let talias = Env.TypeAlias.get env alias_id in 
+    Env.TypeAlias.push env id (L.unloc talias)
```

Comment:
Can’t you call `tt_type` instead of reimplementing it?

Reply:
Done